### PR TITLE
feat: Add bottom sheet with attendee details while check in

### LINKED ISSUE #183: Show attendee information at the time of check in
Show attendee information in a bottom sheet, along with the ticket type and info about date purchased, receipt number etc.

Along with this, show the attendee bottom sheet in the Scan Activity so that organizer does not have to go to attendee page, again and again, to confirm launch the Scan Activity

test

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #183 

- Update libraries to fix rendering bugs
- Add bottom sheet view with details of ticketing
- Add database change listener interface and class
- Decouple toggling of check-in status in its own presenter

Tests coming in another PR as this one got large

device-2017-06-24-063251
device-2017-06-24-063328

### CODE REVIEW

**Comment by mejariamol** [comment]
Please change cross and correct buttons on the bottom sheet. It is confusing. Like for cross button it seems closing the bottom sheet dialog not checking out.

File: app/src/main/java/org/fossasia/openevent/app/utils/ViewUtils.java [hunk 1]
```diff
@@ -7,15 +7,17 @@
 
 public class ViewUtils {
 
-    public static void showView(View view, boolean show) {
-        int mode = View.GONE;
-
+    public static void showInvisibleView(View view, int mode, boolean show) {
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/utils/ViewUtils.java]: Please remove mode from the method name as it is taking mode in parameter. why not same name?

File: app/src/main/java/org/fossasia/openevent/app/common/binding/BindingAdapters.java [hunk 2]
```diff
@@ -38,9 +39,14 @@ public static void bindCircularImage(ImageView imageView, String url) {
...
 
-    @BindingAdapter({"backgroundTint"})
-    public static void setBackgroundTint(View view, @ColorRes int color) {
-        ViewCompat.setBackgroundTintList(view, ColorStateList.valueOf(ContextCompat.getColor(view.getContext(), color)));
+    @BindingAdapter("backgroundTint")
+    public static void setBackgroundTintColor(View view, @ColorInt int color) {
+        ViewCompat.setBackgroundTintList(view, ColorStateList.valueOf(color));
+    }
+
+    @BindingAdapter("srcCompat")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/binding/BindingAdapters.java]: It fails with data binding. This is workaround

File: app/src/main/java/org/fossasia/openevent/app/utils/ViewUtils.java (hunk 1 above)
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/utils/ViewUtils.java]: True

**Comment by iamareebjamal** [comment]
@mejariamol Please suggest better icons. I tried to locate similar icons but they are more like sign out, and then there is no icon to reflect the opposite state

**Comment by iamareebjamal** [comment]
@arp95 @Shailesh351 @rohanagarwal94 Please review

**Comment by mejariamol** [comment]
@iamareebjamal How about this
login-variant
logout-variant

**Comment by rohanagarwal94** [comment]
https://cdn4.iconfinder.com/data/icons/hotel-facilities-2/512/check-out-512.png
https://cdn4.iconfinder.com/data/icons/hotel-facilities-2/512/check-in-512.png I prefer these.

**Comment by arp95** [comment]
@rohanagarwal94 nice icons. prefer using them.
rest looks good.

**Comment by iamareebjamal** [comment]
@rohanagarwal94 I'll try to convert them to vector icons

**Comment by rohanagarwal94** [comment]
@iamareebjamal any significance of the 'COMPLETED' in the attendee details?

**Comment by iamareebjamal** [comment]
It's showing the status of the transaction of order

**Comment by iamareebjamal** [comment]
@rohanagarwal94 This is a commercial copyright icon. Can't use it. Using @mejariamol's icon for now

**Comment by iamareebjamal** [comment]
@mejariamol Any specific source for vector assets for these icons?

**Comment by mejariamol** [comment]
https://materialdesignicons.com/ 

for `login` `logout`

**Comment by iamareebjamal** [comment]
@mejariamol Notice that only the modify attendee is changed in the attendee list behind. This feature shouldn't break in #220

https://streamable.com/s/f4uu2/vdle

**Comment by iamareebjamal** [comment]
@arp95 @Shailesh351 @rohanagarwal94 Please review now

File: app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesFragment.java [hunk 3]
```diff
@@ -148,10 +147,8 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
             return;
         long attendeeId = data.getLongExtra(Constants.SCANNED_ATTENDEE, -1);
 
-        if (attendeeId != -1) {
-            attendeesPresenter.getAttendeeById(attendeeId)
-                .subscribe(this::showToggleDialog, Timber::e);
-        }
+        if (attendeeId != -1)
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesFragment.java]: Now as we got this dialog fragment we shall show this dialog on QR scan activity itself no need to start it for result.

What do you say? 
don't have to do in this PR itself. I will open issue for the same.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesFragment.java]: Yes, that's what should happen :+1:

**Comment by iamareebjamal** [comment]
@niranjan94 Please merge

**Comment by mejariamol** [comment]
@iamareebjamal 
>Notice that only the modify attendee is changed in the attendee list behind. This feature shouldn't break in #220

This won't affect #220 and vice versa.

**Pull request merged**